ouroboros-consensus/test-consensus/Test/ThreadNet/DualPBFT.hs

```diff
@@ -45,12 +46,38 @@ tests = testGroup "DualPBFT" [
 
 prop_convergence :: SetupDualPBft -> Property
 prop_convergence setup =
+    (\prop -> if mightForgeInSlot0 then discard else prop) $
+    tabulate "Ref.PBFT result" [Ref.resultConstrName refResult] $
     prop_general
       (setupSecurityParam      setup)
-      (setupConfig             setup)
+      cfg
       (setupSchedule           setup)
+      (Just $ NumBlocks $ case refResult of
+         Ref.Forked{} -> 1
+         _            -> 0)
       (setupExpectedRejections setup)
       (setupTestOutput         setup)
+  where
+    cfg = setupConfig setup
+
+    refResult :: Ref.Result
+    refResult =
+      Ref.simulate (setupParams setup) (nodeJoinPlan cfg) (numSlots cfg)
+
+    -- The test infrastructure allows nodes to forge in slot 0; however, the
```

Comment:
@edsko: does this make sense? At the low-level mechanics level, this restriction is "due" to `cvsLastSlot :: SlotNo` instead of `cvsLastSlot :: WithOrigin SlotNo`.

Reply:
I see. I think it's okay for now. The treatment of the point near genesis is a mess to be honest, we ought to clean that up at some point.